rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/policy/TimeSlotPolicy.java

```diff
@@ -57,62 +57,37 @@
 @ActiveObject
 public class TimeSlotPolicy extends NodeSourcePolicy implements InitActive {
 
-    /**
-     * Timer task acquired all node from infrastructure
-     */
-    private class AcquireTask extends TimerTask {
-        @Override
-        public void run() {
-            synchronized (timer) {
-                acquireAllNodes();
-            }
-        }
-    }
+    private static final String TIMER_NAME = "TimeSlotPolicy Timer";
 
-    /**
-     * Timer task released all node from infrastructure
-     */
-    private class ReleaseTask extends TimerTask {
-        @Override
-        public void run() {
-            synchronized (timer) {
-                // security trick
-                thisStub.removeAllNodes(preemptive);
-            }
-        }
-    }
-
-    /** Date formatter */
-    public static transient DateFormat dateFormat = SimpleDateFormat.getDateTimeInstance(SimpleDateFormat.SHORT,
-                                                                                         SimpleDateFormat.FULL,
-                                                                                         new Locale("en"));
+    public static final transient DateFormat DATE_FORMAT = SimpleDateFormat.getDateTimeInstance(SimpleDateFormat.SHORT,
+                                                                                                SimpleDateFormat.FULL,
+                                                                                                new Locale("en"));
 
     /** Timer used for tasks scheduling */
-    private transient Timer timer = new Timer("TimeSlotPolicy Timer", true);
+    private transient Timer timer;
 
     /**
      * Initial time for nodes acquisition
      */
-    @Configurable
-    private String acquireTime = dateFormat.format(getDate(System.currentTimeMillis()));
+    @Configurable(description = "Time of nodes acquisition", dynamic = true)
+    private String acquireTime = DATE_FORMAT.format(getDate(System.currentTimeMillis()));
 
     /**
      * Initial time for nodes releasing
      */
-    @Configurable
-    private String releaseTime = dateFormat.format(getDate(System.currentTimeMillis() +
-                                                           1 * 60 * 60 * 1000 /* 1 hour */));
+    @Configurable(description = "Time of nodes removal", dynamic = true)
+    private String releaseTime = DATE_FORMAT.format(getDate(System.currentTimeMillis() + 60 * 60 * 1000)); // 1 hour
 
     /**
      * Period in milliseconds between nodes acquisition/releasing iterations
      */
-    @Configurable(description = "ms (1 day by default)")
-    private Long period = new Long(24 * 60 * 60 * 1000);
+    @Configurable(description = "ms (1 day by default)", dynamic = true)
+    private Long period = (long) (24 * 60 * 60 * 1000);
```

Comment:
Why going from new Long() to (long) ?

Reply:
I changed that into `24 * 60 * 60 * 1000L` which avoids casting and which is more readable in my opinion.

Reply:
Both are ok for me, I was just curious.